packages/client/tests/functional/reuse-binary-engine/tests.ts

```diff
@@ -0,0 +1,70 @@
+import * as util from 'util'
+
+import { setupTestSuiteMatrix } from '../_utils/setupTestSuiteMatrix'
+
+// https://github.com/prisma/prisma/issues/12507
+setupTestSuiteMatrix(() => {
+  test('should create data using one PrismaClient and read using another', async () => {
+    // @ts-ignore
+    const prismaClient1 = new PrismaClient({
+      log: [
+        {
+          emit: 'event',
+          level: 'info',
+        },
+      ],
+    })
+
+    let prismaClient2 = undefined
+    let internalURL = ''
+
+    prismaClient1.$on('info', (data) => {
+      if (/Started query engine/.test(data.message)) {
+        const words = data.message.split(' ')
```

Comment:
I think this could be simplified into:
```
let port = data.message.split(':').pop() // pops off the last one
let URL = `http://127.0.0.1:${port}`
```
```

Reply:
I'd actually suggest `URL` class:

```new URL(url).port`